src/ui/public/visualize/loader/pipeline_helpers/build_pipeline.ts

```diff
@@ -143,7 +144,14 @@ export const getSchemas = (vis: Vis, timeRange?: any): Schemas => {
   };
   const responseAggs = vis.aggs.getResponseAggs().filter((agg: AggConfig) => agg.enabled);
   const isHierarchical = vis.isHierarchical();
+  const minimalColumns = hasMinimalColumns(vis);
   const metrics = responseAggs.filter((agg: AggConfig) => agg.type.type === 'metrics');
+
+  // Some simple math to determine what the column count will be that comes back from tabify.
+  const metricCnt = metrics.length;
```

Comment:
ight now we do this for every visualization .... but this settings only exists for table.

Reply:
On line 239 you could do:
```
if (hasMinimalColumns(vis)) {
  viConfig.dimensions.metrics = [visConfig.dimensions.metrics.pop()];
}
```

Reply:
probably we don't need hasMinimalColumns and could just do:
```
if (vis.params.showMinimalColumns === false) {
  viConfig.dimensions.metrics = [visConfig.dimensions.metrics.pop()];
}
```